pages/auth/redirect.vue

```diff
@@ -18,53 +18,48 @@ const { handleConnectorRedirect } = store
 const { wallet, signer } = storeToRefs(store)
 const { signMessageMemo, disconnect } = store
 const bookStoreApiStore = useBookStoreApiStore()
-const { authenticate } = bookStoreApiStore
-const { token, wallet: sessionWallet } = storeToRefs(bookStoreApiStore)
+const { authenticate, clearSession } = bookStoreApiStore
 
 const route = useRoute()
-const router = useRouter()
 const toast = useToast()
 
 onMounted(async () => {
   const { method, code } = route.query
   if (method && code) {
-    await handleConnectorRedirect({
-      method: method as string,
-      params: { code }
-    })
+    try {
+      await handleConnectorRedirect({
+        method: method as string,
+        params: { code }
+      })
 
-    if (signer.value) {
-      const signature = await signMessageMemo(
-        'authorize',
-        ['read:nftbook', 'write:nftbook', 'read:nftcollection', 'write:nftcollection']
-      )
-      if (!signature) {
-        disconnect()
-        router.replace('/')
-        return
-      }
-      await authenticate(wallet.value, signature)
-      try {
-        window.localStorage.setItem('likecoin_nft_book_press_token', JSON.stringify({ wallet: sessionWallet.value, token: token.value }))
-      } catch (err) {
-        // eslint-disable-next-line no-console
-        console.error(err)
+      if (signer.value) {
+        const signature = await signMessageMemo(
+          'authorize',
+          ['read:nftbook', 'write:nftbook', 'read:nftcollection', 'write:nftcollection']
+        )
+        if (!signature) {
+          disconnect()
+          clearSession()
+        } else {
```

Comment:
Should prefer early return/throw instead of else if the case above is an error guard